homeassistant/components/litterrobot/entity.py

```diff
@@ -36,18 +36,17 @@ class LitterRobotEntity(
 
     _attr_has_entity_name = True
 
-    def __init__(self, robot: _RobotT, entity_type: str, hub: LitterRobotHub) -> None:
+    def __init__(
+        self, robot: _RobotT, hub: LitterRobotHub, description: EntityDescription
+    ) -> None:
         """Pass coordinator to CoordinatorEntity."""
         super().__init__(hub.coordinator)
         self.robot = robot
-        self.entity_type = entity_type
         self.hub = hub
-        self._attr_name = entity_type.capitalize()
-
-    @property
-    def unique_id(self) -> str:
-        """Return a unique ID."""
-        return f"{self.robot.serial}-{self.entity_type}"
+        self.entity_description = description
+        self._attr_unique_id = f"{self.robot.serial}-{description.name}"
+        if description.name is not None:
+            self._attr_name = description.name.capitalize()
```

Comment:
In a follow-up PR, I think it would be good to adjust `self._attr_unique_id` to use the `description.key` instead of `description.name`, and then you can remove these two lines. I think that will require a migration process though...

Reply:
I've got a stashed change for that already and plan on adding it!